File: GitHub/Models/ValidationError_errors.cs
using Microsoft.Kiota.Abstractions.Serialization;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
namespace GithubTodoDemo.GitHub.Models {
    public class ValidationError_errors : IAdditionalDataHolder, IParsable {
        /// <summary>Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.</summary>
        public IDictionary<string, object> AdditionalData { get; set; }
        /// <summary>The code property</summary>
        public string Code { get; set; }
        /// <summary>The field property</summary>
        public string Field { get; set; }
        /// <summary>The index property</summary>
        public int? Index { get; set; }
        /// <summary>The message property</summary>
        public string Message { get; set; }
        /// <summary>The resource property</summary>
        public string Resource { get; set; }
        /// <summary>The value property</summary>
        public Pulls Value { get; set; }
        /// <summary>
        /// Instantiates a new ValidationError_errors and sets the default values.
        /// </summary>
        public ValidationError_errors() {
            AdditionalData = new Dictionary<string, object>();
        }
        /// <summary>
        /// Creates a new instance of the appropriate class based on discriminator value
        /// <param name="parseNode">The parse node to use to read the discriminator value and create the object</param>
        /// </summary>
        public static ValidationError_errors CreateFromDiscriminatorValue(IParseNode parseNode) {
            _ = parseNode ?? throw new ArgumentNullException(nameof(parseNode));
            return new ValidationError_errors();
        }
        /// <summary>
        /// The deserialization information for the current model
        /// </summary>
        public IDictionary<string, Action<IParseNode>> GetFieldDeserializers() {
            return new Dictionary<string, Action<IParseNode>> {
                {"code", n => { Code = n.GetStringValue(); } },
                {"field", n => { Field = n.GetStringValue(); } },
                {"index", n => { Index = n.GetIntValue(); } },
                {"message", n => { Message = n.GetStringValue(); } },
                {"resource", n => { Resource = n.GetStringValue(); } },
                {"value", n => { Value = n.GetObjectValue<Pulls>(Pulls.CreateFromDiscriminatorValue); } },
            };
        }
        /// <summary>
        /// Serializes information the current object
        /// <param name="writer">Serialization writer to use to serialize this model</param>
        /// </summary>
        public void Serialize(ISerializationWriter writer) {
            _ = writer ?? throw new ArgumentNullException(nameof(writer));
            writer.WriteStringValue("code", Code);
            writer.WriteStringValue("field", Field);
            writer.WriteIntValue("index", Index);
            writer.WriteStringValue("message", Message);
            writer.WriteStringValue("resource", Resource);
            writer.WriteObjectValue<Pulls>("value", Value);
            writer.WriteAdditionalData(AdditionalData);
        }
        /// <summary>Composed type wrapper for classes string, integer, string</summary>
        public class Pulls : IAdditionalDataHolder, IParsable {
            /// <summary>Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.</summary>
            public IDictionary<string, object> AdditionalData { get; set; }
            /// <summary>Composed type representation for type integer</summary>
            public int? Integer { get; set; }
            /// <summary>Serialization hint for the current wrapper.</summary>
            public string SerializationHint { get; set; }
            /// <summary>Composed type representation for type string</summary>
            public string String { get; set; }
            /// <summary>
            /// Instantiates a new pulls and sets the default values.
            /// </summary>
            public Pulls() {
                AdditionalData = new Dictionary<string, object>();
            }
            /// <summary>
            /// Creates a new instance of the appropriate class based on discriminator value
            /// <param name="parseNode">The parse node to use to read the discriminator value and create the object</param>
            /// </summary>
            public static Pulls CreateFromDiscriminatorValue(IParseNode parseNode) {
                _ = parseNode ?? throw new ArgumentNullException(nameof(parseNode));
                var mappingValue = parseNode.GetChildNode("")?.GetStringValue();
                var result = new Pulls();
                if(parseNode.GetIntValue() is int integerValue) {
                    result.Integer = integerValue;
                }
                else if(parseNode.GetStringValue() is string stringValue) {
                    result.String = stringValue;
                }
                return result;
            }
            /// <summary>
            /// The deserialization information for the current model
            /// </summary>
            public IDictionary<string, Action<IParseNode>> GetFieldDeserializers() {
                return new Dictionary<string, Action<IParseNode>>();
            }
            /// <summary>
            /// Serializes information the current object
            /// <param name="writer">Serialization writer to use to serialize this model</param>
            /// </summary>
            public void Serialize(ISerializationWriter writer) {
                _ = writer ?? throw new ArgumentNullException(nameof(writer));
                if(Integer != null) {
                    writer.WriteIntValue(null, Integer);
                }
                else if(String != null) {
                    writer.WriteStringValue(null, String);
                }
                writer.WriteAdditionalData(AdditionalData);
            }
        }
    }
}

File: python/githubclient/models/validation_error_errors.py
from __future__ import annotations
from collections.abc import Callable
from dataclasses import dataclass, field
from kiota_abstractions.serialization import AdditionalDataHolder, Parsable, ParseNode, SerializationWriter
from typing import Any, Optional, TYPE_CHECKING, Union

if TYPE_CHECKING:
    from .validation_error_errors_value import ValidationError_errors_value


@dataclass
class ValidationError_errors(AdditionalDataHolder, Parsable):
    # Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
    additional_data: dict[str, Any] = field(default_factory=dict)

    # The code property
    code: Optional[str] = None
    # The field property
    field: Optional[str] = None
    # The index property
    index: Optional[int] = None
    # The message property
    message: Optional[str] = None
    # The resource property
    resource: Optional[str] = None
    # The value property
    value: Optional[ValidationError_errors_value] = None

    @staticmethod
    def create_from_discriminator_value(
            parse_node: ParseNode) -> ValidationError_errors:
        """
        Creates a new instance of the appropriate class based on discriminator value
        param parse_node: The parse node to use to read the discriminator value and create the object
        Returns: ValidationError_errors
        """
        if parse_node is None:
            raise TypeError("parse_node cannot be null.")
        return ValidationError_errors()

    def get_field_deserializers(
        self, ) -> dict[str, Callable[[ParseNode], None]]:
        """
        The deserialization information for the current model
        Returns: dict[str, Callable[[ParseNode], None]]
        """
        from .validation_error_errors_value import ValidationError_errors_value

        from .validation_error_errors_value import ValidationError_errors_value

        fields: dict[str, Callable[[Any], None]] = {
            "code":
            lambda n: setattr(self, 'code', n.get_str_value()),
            "field":
            lambda n: setattr(self, 'field', n.get_str_value()),
            "index":
            lambda n: setattr(self, 'index', n.get_int_value()),
            "message":
            lambda n: setattr(self, 'message', n.get_str_value()),
            "resource":
            lambda n: setattr(self, 'resource', n.get_str_value()),
            "value":
            lambda n: setattr(self, 'value',
                              n.get_object_value(ValidationError_errors_value)
                              ),
        }
        return fields

    def serialize(self, writer: SerializationWriter) -> None:
        """
        Serializes information the current object
        param writer: Serialization writer to use to serialize this model
        Returns: None
        """
        if writer is None:
            raise TypeError("writer cannot be null.")
        writer.write_str_value("code", self.code)
        writer.write_str_value("field", self.field)
        writer.write_int_value("index", self.index)
        writer.write_str_value("message", self.message)
        writer.write_str_value("resource", self.resource)
        writer.write_object_value("value", self.value)
        writer.write_additional_data_value(self.additional_data)
